// @vicoders/reactive-form/esm2015/lib/directives/phone-input/phone-input.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/phone-input/phone-input.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, forwardRef } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { Countries } from './countries';
export class PhoneInputComponent {
    constructor() {
        this._value = {
            code: '',
            value: '',
            alpha2Code: ''
        };
        this.COUNTRIES = Countries;
        this.propagateChange = (/**
         * @param {?} val
         * @return {?}
         */
        (val) => { });
    }
    /**
     * @return {?}
     */
    get value() {
        return this._value;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    set value(val) {
        this._value = val;
        this.propagateChange(this._value);
    }
    /**
     * @return {?}
     */
    ngOnChanges() {
        if (this.value) {
            $('#rf-phone-code')
                .select2()
                .val(this.value.code + '-' + this.value.alpha2Code)
                .trigger('change.select2');
        }
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        /** @type {?} */
        const $this = this;
        setTimeout((/**
         * @return {?}
         */
        () => {
            $('#rf-phone-code').select2();
            if (this.value) {
                $('#rf-phone-code')
                    .select2()
                    .val(this.value.code + '-' + this.value.alpha2Code)
                    .trigger('change.select2');
            }
            $('#rf-phone-code').change((/**
             * @return {?}
             */
            function () {
                // tslint:disable-next-line:prefer-const
                /** @type {?} */
                const val = $(this)
                    .val()
                    .split('-');
                $this.value = Object.assign({}, $this.value, { code: val[0], alpha2Code: val[1] });
            }));
        }), 1000);
    }
    /**
     * @param {?} val
     * @return {?}
     */
    selectCountry(val) {
        val = val.split('-');
        this.value = Object.assign({}, this.value, { code: val[0], alpha2Code: val[1] });
    }
    /**
     * @param {?} val
     * @return {?}
     */
    enterPhone(val) {
        this.value = Object.assign({}, this.value, { value: val });
    }
    /**
     * @param {?} value
     * @return {?}
     */
    writeValue(value) {
        if (value !== undefined) {
            this.value = value;
        }
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnChange(fn) {
        this.propagateChange = fn;
    }
    /**
     * @return {?}
     */
    registerOnTouched() { }
}
PhoneInputComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'app-phone-input',
                template: "<div class=\"phone-box form-group\" id=\"rf-phone-code-group\">\n  <select\n    id=\"rf-phone-code\"\n    [attr.disabled]=\"isDisabled\"\n    #country\n    (change)=\"selectCountry(country.value)\"\n    class=\"phone-code\"\n    [(ngModel)]=\"value.code + '-' + value.alpha2Code\"\n    select2\n  >\n    <option value=\"{{ country.callingCodes[0] }}-{{ country.alpha2Code }}\" *ngFor=\"let country of COUNTRIES\">\n      <span>{{ country.name }}</span>\n    </option>\n  </select>\n  <div class=\"phone-number-box\" [ngClass]=\"{ disabled: isDisabled }\">\n    <span class=\"plus-code\" *ngIf=\"value.code\">(+{{ value.code }}) </span>\n    <input\n      [attr.disabled]=\"isDisabled\"\n      type=\"text\"\n      #phone\n      (change)=\"enterPhone(phone.value)\"\n      class=\"phone-value\"\n      [(ngModel)]=\"value.value\"\n    />\n  </div>\n</div>\n",
                providers: [
                    {
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef((/**
                         * @return {?}
                         */
                        () => PhoneInputComponent)),
                        multi: true
                    }
                ],
                styles: [".phone-box{position:relative;clear:both;border:1px solid #ccc;height:31px}.phone-box .phone-code{-webkit-appearance:none;-moz-appearance:none;appearance:none;border:0;background-color:#333;color:#fff;border-radius:0;float:left;padding:5px;margin-right:10px;width:120px}.phone-box .phone-number-box{line-height:27px}.phone-box .phone-number-box .phone-value{border:0;background-color:transparent;width:125px}.phone-box .disabled{background-color:#e9ecef}::ng-deep #rf-phone-code-group .select2-container{float:left}::ng-deep #rf-phone-code-group .plus-code{margin-left:12px}::ng-deep .select2-container .select2-selection--single{height:29px;border-radius:0;border:none;border-right:1px solid #ccc}"]
            }] }
];
PhoneInputComponent.propDecorators = {
    isDisabled: [{ type: Input }],
    _value: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    PhoneInputComponent.prototype.isDisabled;
    /** @type {?} */
    PhoneInputComponent.prototype._value;
    /** @type {?} */
    PhoneInputComponent.prototype.COUNTRIES;
    /** @type {?} */
    PhoneInputComponent.prototype.propagateChange;
}
