// @vicoders/reactive-form/esm2015/lib/models/DateTimePicker.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/models/DateTimePicker.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InputBase } from './InputBase';
export class DateTimePicker extends InputBase {
    /**
     * @param {?=} options
     */
    constructor(options = {}) {
        super(options);
        this.controlType = 'datetimepicker';
        this.showIcon = options['showIcon'] || false;
        this.monthNavigator = options['monthNavigator'] || false;
        this.yearNavigator = options['yearNavigator'] || false;
        this.yearRange = options['yearRange'] || '1995:2050';
        this.showTime = options['showTime'] || false;
        this.timeOnly = options['timeOnly'] || false;
        this.dateFormat = options['dateFormat'] || 'dd/mm/yy';
        this.minDate = options['minDate'] || null;
        this.maxDate = options['maxDate'] || null;
    }
}
if (false) {
    /** @type {?} */
    DateTimePicker.prototype.controlType;
    /** @type {?} */
    DateTimePicker.prototype.showIcon;
    /** @type {?} */
    DateTimePicker.prototype.monthNavigator;
    /** @type {?} */
    DateTimePicker.prototype.yearNavigator;
    /** @type {?} */
    DateTimePicker.prototype.yearRange;
    /** @type {?} */
    DateTimePicker.prototype.showTime;
    /** @type {?} */
    DateTimePicker.prototype.timeOnly;
    /** @type {?} */
    DateTimePicker.prototype.dateFormat;
    /** @type {?} */
    DateTimePicker.prototype.minDate;
    /** @type {?} */
    DateTimePicker.prototype.maxDate;
}
