// @vicoders/reactive-form/esm2015/lib/models/DropZone.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/models/DropZone.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InputBase } from './InputBase';
export class DropZone extends InputBase {
    /**
     * @param {?=} options
     */
    constructor(options = {}) {
        super(options);
        this.controlType = 'dropzone';
        this.message = options['message'] || '';
        this.upload_path = options['upload_path'] || '';
        this.acceptedFiles = options['acceptedFiles'] || '*';
        this.maxFilesize = options['maxFilesize'] || 50;
        this.maxFiles = options['maxFiles'] || 0;
        this.showPreview = options['showPreview'] || false;
        this.url = options['url'] || '';
        this.headers = options['headers'] || null;
        this.paramName = options['paramName'] || 'file';
        this.resultTransformer = options['resultTransformer'];
    }
}
if (false) {
    /** @type {?} */
    DropZone.prototype.controlType;
    /** @type {?} */
    DropZone.prototype.message;
    /** @type {?} */
    DropZone.prototype.upload_path;
    /** @type {?} */
    DropZone.prototype.acceptedFiles;
    /** @type {?} */
    DropZone.prototype.maxFilesize;
    /** @type {?} */
    DropZone.prototype.maxFiles;
    /** @type {?} */
    DropZone.prototype.showPreview;
    /** @type {?} */
    DropZone.prototype.url;
    /** @type {?} */
    DropZone.prototype.headers;
    /** @type {?} */
    DropZone.prototype.paramName;
    /** @type {?} */
    DropZone.prototype.resultTransformer;
}
