// @vicoders/reactive-form/esm2015/lib/models/RangerSlider.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/models/RangerSlider.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InputBase } from './InputBase';
export class RangerSlider extends InputBase {
    /**
     * @param {?=} options
     */
    constructor(options = {}) {
        super(options);
        this.controlType = 'ranger-slider';
        this.id = options['id'] || 1;
        this.min = options['min'] || 0;
        this.max = options['max'] || 10;
        this.step = options['step'] || 1;
        this.miLabel = options['miLabel'] || '';
        this.unit = options['unit'] || '';
    }
}
if (false) {
    /** @type {?} */
    RangerSlider.prototype.controlType;
    /** @type {?} */
    RangerSlider.prototype.id;
    /** @type {?} */
    RangerSlider.prototype.min;
    /** @type {?} */
    RangerSlider.prototype.max;
    /** @type {?} */
    RangerSlider.prototype.step;
    /** @type {?} */
    RangerSlider.prototype.miLabel;
    /** @type {?} */
    RangerSlider.prototype.unit;
}
