// @vicoders/reactive-form/esm2015/lib/models/Select2.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/models/Select2.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InputBase } from './InputBase';
export class Select2 extends InputBase {
    /**
     * @param {?=} options
     */
    constructor(options = {}) {
        super(options);
        this.controlType = 'select2';
        this.options = [];
        this.id = options['id'] || 1;
        this.options = options['options'] || [];
        this.tags = options['tags'] || false;
        this.placeholder = options['placeholder'] || '';
        this.isSelectAll = options['isSelectAll'] || false;
        this.selectBtnText = options['selectBtnText'] || 'Select All';
        this.dropdownParent = options['dropdownParent'];
        this.clearBtnText = options['clearBtnText'] || 'Clear All';
    }
}
if (false) {
    /** @type {?} */
    Select2.prototype.controlType;
    /** @type {?} */
    Select2.prototype.options;
    /** @type {?} */
    Select2.prototype.tags;
    /** @type {?} */
    Select2.prototype.placeholder;
    /** @type {?} */
    Select2.prototype.id;
    /** @type {?} */
    Select2.prototype.isSelectAll;
    /** @type {?} */
    Select2.prototype.selectBtnText;
    /** @type {?} */
    Select2.prototype.clearBtnText;
    /** @type {?} */
    Select2.prototype.dropdownParent;
}
