// @vicoders/reactive-form/esm2015/lib/models/SingleSelect2.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/models/SingleSelect2.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InputBase } from './InputBase';
export class SingleSelect2 extends InputBase {
    /**
     * @param {?=} options
     */
    constructor(options = {}) {
        super(options);
        this.controlType = 'single-select2';
        this.options = [];
        this.id = options['id'] || 1;
        this.options = options['options'] || [];
        this.placeholder = options['placeholder'] || '';
        this.dropdownParent = options['dropdownParent'];
    }
}
if (false) {
    /** @type {?} */
    SingleSelect2.prototype.controlType;
    /** @type {?} */
    SingleSelect2.prototype.options;
    /** @type {?} */
    SingleSelect2.prototype.placeholder;
    /** @type {?} */
    SingleSelect2.prototype.id;
    /** @type {?} */
    SingleSelect2.prototype.dropdownParent;
}
