// @vicoders/reactive-form/esm2015/lib/models/TinyMce.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/models/TinyMce.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InputBase } from './InputBase';
export class TinyMce extends InputBase {
    /**
     * @param {?=} options
     */
    constructor(options = {}) {
        super(options);
        this.controlType = 'tinymce';
        this.type = options['type'] || '';
        this.height = options['height'] || '300';
        this.plugins = options['plugins'] || '';
        this.apiUpload = options['apiUpload'] || '';
        this.paramName = options['paramName'] || 'files';
        this.resultTransformer = options['resultTransformer'];
    }
}
if (false) {
    /** @type {?} */
    TinyMce.prototype.controlType;
    /** @type {?} */
    TinyMce.prototype.type;
    /** @type {?} */
    TinyMce.prototype.height;
    /** @type {?} */
    TinyMce.prototype.plugins;
    /** @type {?} */
    TinyMce.prototype.apiUpload;
    /** @type {?} */
    TinyMce.prototype.paramName;
    /** @type {?} */
    TinyMce.prototype.resultTransformer;
}
