// @vicoders/reactive-form/esm2015/lib/models/TypeAhead.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/models/TypeAhead.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InputBase } from './InputBase';
export class TypeAhead extends InputBase {
    /**
     * @param {?=} options
     */
    constructor(options = {}) {
        super(options);
        this.controlType = 'type-ahead';
        this.resultTransformer = options['resultTransformer'] || null;
        this.apiUpload = options['apiUpload'] || '';
        this.method = options['method'] || 'get';
        this.keyName = options['keyName'] || null;
        this.fieldName = options['fieldName'];
        this.id = options['id'] || 1;
        this.searchBy = options['searchBy'];
    }
}
if (false) {
    /** @type {?} */
    TypeAhead.prototype.controlType;
    /** @type {?} */
    TypeAhead.prototype.resultTransformer;
    /** @type {?} */
    TypeAhead.prototype.apiUpload;
    /** @type {?} */
    TypeAhead.prototype.method;
    /** @type {?} */
    TypeAhead.prototype.keyName;
    /** @type {?} */
    TypeAhead.prototype.fieldName;
    /** @type {?} */
    TypeAhead.prototype.id;
    /** @type {?} */
    TypeAhead.prototype.searchBy;
}
