// @vicoders/reactive-form/esm2015/lib/models/TypeAheadWithoutApi.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/models/TypeAheadWithoutApi.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InputBase } from './InputBase';
export class TypeAheadWithoutApi extends InputBase {
    /**
     * @param {?=} options
     */
    constructor(options = {}) {
        super(options);
        this.controlType = 'type-ahead-without-api';
        this.options = options['options'] || [];
        this.fieldName = options['fieldName'] || [];
        this.id = options['id'] || 1;
        this.searchBy = options['searchBy'] || [];
    }
}
if (false) {
    /** @type {?} */
    TypeAheadWithoutApi.prototype.controlType;
    /** @type {?} */
    TypeAheadWithoutApi.prototype.options;
    /** @type {?} */
    TypeAheadWithoutApi.prototype.fieldName;
    /** @type {?} */
    TypeAheadWithoutApi.prototype.id;
    /** @type {?} */
    TypeAheadWithoutApi.prototype.searchBy;
}
