// @vicoders/reactive-form/esm2015/lib/models/UploadFileV2.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/models/UploadFileV2.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InputBase } from './InputBase';
export class UploadFileV2 extends InputBase {
    /**
     * @param {?=} options
     */
    constructor(options = {}) {
        super(options);
        this.controlType = 'uploadfilev2';
        this.id = options['id'] || 1;
        this.accept = options['accept'] || '';
        this.allowMaxSize = options['allowMaxSize'] || 2;
        this.apiUpload = options['apiUpload'] || '';
        this.headers = options['headers'] || {};
        this.uploadPath = options['uploadPath'] || '';
        this.multiple = options['multiple'] || false;
        this.paramName = options['paramName'] || 'files';
        this.resultTransformer = options['resultTransformer'];
        this.title = options['title'] || 'Upload';
        this.dataFunction = options['dataFunction'];
        this.uploadFromDB = options['uploadFromDB'] || false;
        this.width = options['width'];
        this.maxWidth = options['maxWidth'];
        this.height = options['height'];
        this.maxHeight = options['maxHeight'];
    }
}
if (false) {
    /** @type {?} */
    UploadFileV2.prototype.controlType;
    /** @type {?} */
    UploadFileV2.prototype.id;
    /** @type {?} */
    UploadFileV2.prototype.accept;
    /** @type {?} */
    UploadFileV2.prototype.allowMaxSize;
    /** @type {?} */
    UploadFileV2.prototype.apiUpload;
    /** @type {?} */
    UploadFileV2.prototype.headers;
    /** @type {?} */
    UploadFileV2.prototype.uploadPath;
    /** @type {?} */
    UploadFileV2.prototype.multiple;
    /** @type {?} */
    UploadFileV2.prototype.paramName;
    /** @type {?} */
    UploadFileV2.prototype.resultTransformer;
    /** @type {?} */
    UploadFileV2.prototype.title;
    /** @type {?} */
    UploadFileV2.prototype.dataFunction;
    /** @type {?} */
    UploadFileV2.prototype.uploadFromDB;
    /** @type {?} */
    UploadFileV2.prototype.width;
    /** @type {?} */
    UploadFileV2.prototype.maxWidth;
    /** @type {?} */
    UploadFileV2.prototype.height;
    /** @type {?} */
    UploadFileV2.prototype.maxHeight;
}
