// @vicoders/reactive-form/esm2015/lib/models/UploadFileV3.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/models/UploadFileV3.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { InputBase } from './InputBase';
export class UploadFileV3 extends InputBase {
    /**
     * @param {?=} options
     */
    constructor(options = {}) {
        super(options);
        this.controlType = 'uploadfilev3';
        this.id = options['id'] || 1;
        this.accept = options['accept'] || '';
        this.allowMaxSize = options['allowMaxSize'] || 2;
        this.apiUpload = options['apiUpload'] || '';
        this.headers = options['headers'] || {};
        this.uploadPath = options['uploadPath'] || '';
        this.multiple = options['multiple'] || false;
        this.paramName = options['paramName'] || 'files';
        this.resultTransformer = options['resultTransformer'];
        this.resultKey = options['resultKey'];
    }
}
if (false) {
    /** @type {?} */
    UploadFileV3.prototype.controlType;
    /** @type {?} */
    UploadFileV3.prototype.id;
    /** @type {?} */
    UploadFileV3.prototype.accept;
    /** @type {?} */
    UploadFileV3.prototype.allowMaxSize;
    /** @type {?} */
    UploadFileV3.prototype.apiUpload;
    /** @type {?} */
    UploadFileV3.prototype.headers;
    /** @type {?} */
    UploadFileV3.prototype.uploadPath;
    /** @type {?} */
    UploadFileV3.prototype.multiple;
    /** @type {?} */
    UploadFileV3.prototype.paramName;
    /** @type {?} */
    UploadFileV3.prototype.resultTransformer;
    /** @type {?} */
    UploadFileV3.prototype.resultKey;
}
