// @vicoders/reactive-form/esm2015/lib/directives/draft-js/draft-js.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/draft-js/draft-js.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef } from '@angular/core';
// import { DraftJs } from '@lebaotrung/react_draft-js';
export class DraftJsComponent {
    /**
     * @param {?} el
     */
    constructor(el) {
        this.options = {};
        this.el = el;
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        // tslint:disable-next-line:no-unused-expression
        // new DraftJs(this.el.nativeElement, this.options);
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    ngOnChanges(changes) {
        // console.log(new DraftJs(this.el.nativeElement, this.options));
    }
}
DraftJsComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'draft-js',
                template: "<p>\n  draft-js works!\n</p>\n",
                styles: [":host ::ng-deep .headline{text-align:center;background:-webkit-gradient(linear,left bottom,left top,from(#ff5370),to(#ff595a));background:linear-gradient(to top,#ff5370,#ff595a);font-size:110%;color:#fff;padding:.8em 0}:host ::ng-deep .pageComponents{display:-webkit-inline-box;display:inline-flex;overflow:hidden;height:100%;position:fixed;width:100%;-webkit-box-orient:horizontal;-webkit-box-direction:normal;flex-direction:row;justify-content:space-around}:host ::ng-deep .noteListContainer{position:relative;margin:1em 0;padding:1em 0;flex-basis:30%;text-align:left;-webkit-box-ordinal-group:2;order:1;flex-shrink:1}:host ::ng-deep .note-list-header{font-size:20px;text-align:center;padding:1.5em 1em;margin:.75em;color:#585656;position:-webkit-sticky;position:sticky;top:0;-webkit-box-align:center;align-items:center}:host ::ng-deep .note-list{position:relative;overflow-y:auto;overflow-x:hidden;height:68vh;margin:.5em 0 .5em .5em}:host ::ng-deep .note-list-item{padding:1.1em;font-size:13px;letter-spacing:.5px;border-bottom:1px solid #f5efef;min-height:24px;display:-webkit-box;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;-webkit-box-pack:end;justify-content:flex-end;color:#484545}:host ::ng-deep .note-list-item:hover{background:#fbf1f1}:host ::ng-deep .aboveEditor{font-size:1.25em;text-align:left}:host ::ng-deep input[type=text].noteTitle{font-size:14.5px;line-height:20px;height:20px;letter-spacing:.7px;font-family:'Open Sans';border-bottom:2px solid #f6f7fb;border-top:none;border-left:none;border-right:none;width:80%;position:inline;padding:.5em;margin-left:.6em}:host ::ng-deep .tool-bar{border:2px solid #f6f7fb;text-align:center;font-family:'IBM Plex Sans',sans-serif;background-color:#dedcdc;margin:.5em .5em 0;padding:.5em;border-radius:4px;box-shadow:0 0 5px rgba(0,0,0,.1)}:host ::ng-deep .editors{border:none;border-radius:4px;padding:1em 1.5em 3em;margin:-.2em .6em 1em;font-family:Helvetica;line-height:1.4;color:#2c3a51;text-align:left;background-color:#fff;box-shadow:0 0 5px rgba(0,0,0,.1);min-height:36%;overflow-x:hidden;overflow-y:auto;position:relative;height:40vh;font-size:90%}:host ::ng-deep .editors a{color:#00f}:host ::ng-deep a{color:#00f}:host ::ng-deep a:visited{color:#00f}:host ::ng-deep img{min-width:100px;max-width:300px;text-align:center}:host ::ng-deep button#new{padding:.25em;border-radius:100px;width:2em}:host ::ng-deep button.submitNote{border-radius:10px;font-size:13px;float:right;display:inline;padding:.5em;color:#fff;border:none;background-color:#ff5370;margin:.35em}:host ::ng-deep .code-block,div.code-block{color:#90caf9;line-height:1.2;padding:3px;margin-left:4px;background-color:#000;font-family:'Andale Mono';font-size:13px;box-sizing:border-box;white-space:pre-wrap;display:block}#link_url,.RichEditor-styleButton,:host ::ng-deep .styleButton{padding:6px 12px;background:#fff;border:none;color:#ff595a;margin:.35em;max-width:35px;min-width:30px;font-family:'IBM Plex Sans',sans-serif;font-size:13px;height:2.25em}:host ::ng-deep .RichEditor-activeButton{background-color:#feb47b;color:#fff}:host ::ng-deep select{padding:.5em;background:#fff;border:none;color:#ff595a;margin:.35em;border-radius:6px;font-family:'IBM Plex Sans',sans-serif;font-size:13px;height:2.25em}#italic,#underline,:host ::ng-deep #bold{width:30px}:host ::ng-deep #code{font-family:'Andale Mono',monospace;padding-left:3px!important;padding-right:3px!important}:host ::ng-deep .styleButton#bold{font-weight:800}:host ::ng-deep .styleButton#italic{font-style:italic;font-family:'IBM Plex Sans',sans-serif}:host ::ng-deep .styleButton#underline{text-decoration:underline}:host ::ng-deep .material-icons{font-size:13px}"]
            }] }
];
/** @nocollapse */
DraftJsComponent.ctorParameters = () => [
    { type: ElementRef }
];
if (false) {
    /**
     * @type {?}
     * @private
     */
    DraftJsComponent.prototype.el;
    /** @type {?} */
    DraftJsComponent.prototype.options;
}
