// @vicoders/reactive-form/esm5/lib/directives/captcha/captcha.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/captcha/captcha.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
var CaptchaComponent = /** @class */ (function () {
    function CaptchaComponent() {
        this._captchaData = {
            value: '',
            code: ''
        };
        this.propagateChange = (/**
         * @param {?} val
         * @return {?}
         */
        function (val) { });
    }
    /**
     * @return {?}
     */
    CaptchaComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    /**
     * @return {?}
     */
    CaptchaComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        setTimeout((/**
         * @return {?}
         */
        function () {
            _this.captchaData = tslib_1.__assign({}, _this.captchaData, { code: _this.random() });
        }), 1);
    };
    Object.defineProperty(CaptchaComponent.prototype, "captchaData", {
        get: /**
         * @return {?}
         */
        function () {
            return this._captchaData;
        },
        set: /**
         * @param {?} val
         * @return {?}
         */
        function (val) {
            this._captchaData = val;
            this.propagateChange(this._captchaData);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} value
     * @return {?}
     */
    CaptchaComponent.prototype.writeValue = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        if (value !== undefined) {
            this.captchaData = value;
        }
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    CaptchaComponent.prototype.registerOnChange = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.propagateChange = fn;
    };
    /**
     * @return {?}
     */
    CaptchaComponent.prototype.registerOnTouched = /**
     * @return {?}
     */
    function () { };
    /**
     * @return {?}
     */
    CaptchaComponent.prototype.randIntWithZero = /**
     * @return {?}
     */
    function () {
        return Math.floor(Math.random() * 10);
    };
    /**
     * @return {?}
     */
    CaptchaComponent.prototype.random = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var code = '';
        /** @type {?} */
        var characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
        if (this.type === 'number') {
            characters = '0123456789';
            // for (let i = 0; i < this.numberOfChar; i++) {
            //   code += this.randIntWithZero();
            // }
        }
        else if (this.type === 'alphabet') {
            characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
        }
        /** @type {?} */
        var charactersLength = characters.length;
        for (var i = 0; i < this.numberOfChar; i++) {
            code += characters.charAt(Math.floor(Math.random() * charactersLength));
        }
        return code;
    };
    /**
     * @return {?}
     */
    CaptchaComponent.prototype.randomCode = /**
     * @return {?}
     */
    function () {
        this.captchaData = tslib_1.__assign({}, this.captchaData, { code: this.random() });
    };
    CaptchaComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'captcha',
                    template: "<div class=\"captch--rform\">\n  <input class=\"form-control captch--controll\" maxlength=\"{{ numberOfChar }}\" type=\"text\" [(ngModel)]=\"captchaData.value\" />\n  <span *ngIf=\"captchaData.code\">{{ captchaData.code }}</span>\n  <button type=\"button\" (click)=\"randomCode()\"><i class=\"fa fa-refresh\" aria-hidden=\"true\"></i></button>\n  <p class=\"rf__captcha--error\" *ngIf=\"!captchaCorrect\">{{ message }}</p>\n</div>\n",
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef((/**
                             * @return {?}
                             */
                            function () { return CaptchaComponent; })),
                            multi: true
                        }
                    ],
                    styles: [".captch--rform span{font-family:lato;font-weight:700;letter-spacing:2px;font-style:italic;margin:0 15px}.captch--rform button{cursor:pointer}.captch--rform .captch--controll{display:inline-block;width:200px}.captch--rform .rf__captcha--error{margin-top:15px;color:#e74c3c;font-size:15px}"]
                }] }
    ];
    /** @nocollapse */
    CaptchaComponent.ctorParameters = function () { return []; };
    CaptchaComponent.propDecorators = {
        _captchaData: [{ type: Input }],
        numberOfChar: [{ type: Input }],
        type: [{ type: Input }],
        captchaCorrect: [{ type: Input }],
        message: [{ type: Input }]
    };
    return CaptchaComponent;
}());
export { CaptchaComponent };
if (false) {
    /** @type {?} */
    CaptchaComponent.prototype._captchaData;
    /** @type {?} */
    CaptchaComponent.prototype.numberOfChar;
    /** @type {?} */
    CaptchaComponent.prototype.type;
    /** @type {?} */
    CaptchaComponent.prototype.captchaCorrect;
    /** @type {?} */
    CaptchaComponent.prototype.message;
    /** @type {?} */
    CaptchaComponent.prototype.propagateChange;
}
