// @vicoders/reactive-form/esm2015/lib/validators/default.validator.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/validators/default.validator.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { HttpClient } from '@angular/common/http';
import { catchError, tap } from 'rxjs/operators';
import { Injectable } from '@angular/core';
import { AppInjector } from '../app-injector';
class ValidatorService {
    constructor() {
        this.http = AppInjector.get(HttpClient);
    }
    /**
     * @param {?} api
     * @param {?} headers
     * @return {?}
     */
    apiService(api, headers) {
        return this.http.get(api, { headers: headers }).pipe(tap((/**
         * @param {?} result
         * @return {?}
         */
        result => { })), catchError((/**
         * @param {?} error
         * @return {?}
         */
        error => {
            throw error;
        })));
    }
}
ValidatorService.decorators = [
    { type: Injectable }
];
/** @nocollapse */
ValidatorService.ctorParameters = () => [];
if (false) {
    /** @type {?} */
    ValidatorService.prototype.http;
}
class Validators {
    constructor() { }
    /**
     * @param {?} matchValue
     * @return {?}
     */
    isEqualValidator(matchValue) {
        return (/**
         * @param {?} control
         * @return {?}
         */
        (control) => {
            /** @type {?} */
            const value = control.value;
            if (value) {
                /** @type {?} */
                const compareValue = control.root.get(matchValue).value;
                if (value !== compareValue) {
                    return { notEqual: { valid: false, value: control.value } };
                }
            }
            return null;
        });
    }
    /**
     * @param {?} matchValue
     * @return {?}
     */
    isDifferentValidator(matchValue) {
        return (/**
         * @param {?} control
         * @return {?}
         */
        (control) => {
            /** @type {?} */
            const value = control.value;
            if (value) {
                /** @type {?} */
                const compareValue = control.root.get(matchValue).value;
                if (value === compareValue) {
                    return { notDifferent: { valid: false, value: control.value } };
                }
            }
            return null;
        });
    }
    /**
     * @param {?} url
     * @param {?} message
     * @return {?}
     */
    isExistValidator(url, message) {
        return (/**
         * @param {?} control
         * @return {?}
         */
        (control) => {
            /** @type {?} */
            const api = new ValidatorService();
            /** @type {?} */
            const value = control.value;
            /** @type {?} */
            const UPLOAD_DIRECTIVE_API_URL = AppInjector.get('UPLOAD_DIRECTIVE_API_URL');
            /** @type {?} */
            const UPLOAD_DIRECTIVE_HEADERS = AppInjector.get('UPLOAD_DIRECTIVE_HEADERS');
            return new Promise((/**
             * @param {?} resolve
             * @param {?} reject
             * @return {?}
             */
            (resolve, reject) => tslib_1.__awaiter(this, void 0, void 0, function* () {
                if (value) {
                    /** @type {?} */
                    const result = yield api.apiService(UPLOAD_DIRECTIVE_API_URL + url + value, UPLOAD_DIRECTIVE_HEADERS).toPromise();
                    if (result.exist === true) {
                        resolve({ isExist: { valid: false, value: control.value, message: message } });
                    }
                    resolve(null);
                }
                resolve(null);
            })));
        });
    }
}
Validators.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
Validators.ctorParameters = () => [];
/** @type {?} */
export const DefaultValidators = new Validators();
