// @vicoders/reactive-form/esm5/lib/validators/default.validator.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/validators/default.validator.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { HttpClient } from '@angular/common/http';
import { catchError, tap } from 'rxjs/operators';
import { Injectable } from '@angular/core';
import { AppInjector } from '../app-injector';
var ValidatorService = /** @class */ (function () {
    function ValidatorService() {
        this.http = AppInjector.get(HttpClient);
    }
    /**
     * @param {?} api
     * @param {?} headers
     * @return {?}
     */
    ValidatorService.prototype.apiService = /**
     * @param {?} api
     * @param {?} headers
     * @return {?}
     */
    function (api, headers) {
        return this.http.get(api, { headers: headers }).pipe(tap((/**
         * @param {?} result
         * @return {?}
         */
        function (result) { })), catchError((/**
         * @param {?} error
         * @return {?}
         */
        function (error) {
            throw error;
        })));
    };
    ValidatorService.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    ValidatorService.ctorParameters = function () { return []; };
    return ValidatorService;
}());
if (false) {
    /** @type {?} */
    ValidatorService.prototype.http;
}
var Validators = /** @class */ (function () {
    function Validators() {
    }
    /**
     * @param {?} matchValue
     * @return {?}
     */
    Validators.prototype.isEqualValidator = /**
     * @param {?} matchValue
     * @return {?}
     */
    function (matchValue) {
        return (/**
         * @param {?} control
         * @return {?}
         */
        function (control) {
            /** @type {?} */
            var value = control.value;
            if (value) {
                /** @type {?} */
                var compareValue = control.root.get(matchValue).value;
                if (value !== compareValue) {
                    return { notEqual: { valid: false, value: control.value } };
                }
            }
            return null;
        });
    };
    /**
     * @param {?} matchValue
     * @return {?}
     */
    Validators.prototype.isDifferentValidator = /**
     * @param {?} matchValue
     * @return {?}
     */
    function (matchValue) {
        return (/**
         * @param {?} control
         * @return {?}
         */
        function (control) {
            /** @type {?} */
            var value = control.value;
            if (value) {
                /** @type {?} */
                var compareValue = control.root.get(matchValue).value;
                if (value === compareValue) {
                    return { notDifferent: { valid: false, value: control.value } };
                }
            }
            return null;
        });
    };
    /**
     * @param {?} url
     * @param {?} message
     * @return {?}
     */
    Validators.prototype.isExistValidator = /**
     * @param {?} url
     * @param {?} message
     * @return {?}
     */
    function (url, message) {
        var _this = this;
        return (/**
         * @param {?} control
         * @return {?}
         */
        function (control) {
            /** @type {?} */
            var api = new ValidatorService();
            /** @type {?} */
            var value = control.value;
            /** @type {?} */
            var UPLOAD_DIRECTIVE_API_URL = AppInjector.get('UPLOAD_DIRECTIVE_API_URL');
            /** @type {?} */
            var UPLOAD_DIRECTIVE_HEADERS = AppInjector.get('UPLOAD_DIRECTIVE_HEADERS');
            return new Promise((/**
             * @param {?} resolve
             * @param {?} reject
             * @return {?}
             */
            function (resolve, reject) { return tslib_1.__awaiter(_this, void 0, void 0, function () {
                var result;
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            if (!value) return [3 /*break*/, 2];
                            return [4 /*yield*/, api.apiService(UPLOAD_DIRECTIVE_API_URL + url + value, UPLOAD_DIRECTIVE_HEADERS).toPromise()];
                        case 1:
                            result = _a.sent();
                            if (result.exist === true) {
                                resolve({ isExist: { valid: false, value: control.value, message: message } });
                            }
                            resolve(null);
                            _a.label = 2;
                        case 2:
                            resolve(null);
                            return [2 /*return*/];
                    }
                });
            }); }));
        });
    };
    Validators.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    Validators.ctorParameters = function () { return []; };
    return Validators;
}());
/** @type {?} */
export var DefaultValidators = new Validators();
