// @vicoders/reactive-form/esm5/lib/directives/ranger-silders/ranger-silders.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/ranger-silders/ranger-silders.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, forwardRef } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
var RangerSildersComponent = /** @class */ (function () {
    function RangerSildersComponent() {
        this._value = {
            min: null,
            max: null
        };
        this.propagateChange = (/**
         * @param {?} val
         * @return {?}
         */
        function (val) { });
    }
    /**
     * @return {?}
     */
    RangerSildersComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        setTimeout((/**
         * @return {?}
         */
        function () {
            if (_this.oldValue) {
                _this.value = _this.oldValue;
            }
            else {
                _this.value = {
                    min: _this.min,
                    max: _this.max
                };
            }
            /** @type {?} */
            var $this = _this;
            /**
             * @return {?}
             */
            function getVals() {
                // Get slider values
                /** @type {?} */
                var parent = this.parentNode;
                /** @type {?} */
                var slides = parent.getElementsByTagName('input');
                /** @type {?} */
                var slide1 = parseFloat(slides[0].value);
                /** @type {?} */
                var slide2 = parseFloat(slides[1].value);
                // Neither slider will clip the other, so make sure we determine which is larger
                if (slide1 > slide2) {
                    /** @type {?} */
                    var tmp = slide2;
                    slide2 = slide1;
                    slide1 = tmp;
                }
                /** @type {?} */
                var miLabel = $this.miLabel ? $this.miLabel + ': ' : '';
                /** @type {?} */
                var unit = $this.unit ? $this.unit : '';
                /** @type {?} */
                var displayElement = parent.getElementsByClassName('rangeValues')[0];
                displayElement.innerHTML = miLabel + slide1 + unit + ' - ' + slide2 + unit;
                $this.value = {
                    min: slide1,
                    max: slide2
                };
            }
            window.onload = (/**
             * @return {?}
             */
            function () {
                // Initialize Sliders
                /** @type {?} */
                var sliderSections = document.getElementsByClassName('range-slider');
                for (var x = 0; x < sliderSections.length; x++) {
                    /** @type {?} */
                    var sliders = sliderSections[x].getElementsByTagName('input');
                    for (var y = 0; y < sliders.length; y++) {
                        if (sliders[y].type === 'range') {
                            sliders[y].oninput = getVals;
                            // Manually trigger event first time to display values
                            sliders[y].oninput(null);
                        }
                    }
                }
            });
        }), 1000);
    };
    Object.defineProperty(RangerSildersComponent.prototype, "value", {
        get: /**
         * @return {?}
         */
        function () {
            return this._value;
        },
        set: /**
         * @param {?} val
         * @return {?}
         */
        function (val) {
            this._value = val;
            this.propagateChange(this._value);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} value
     * @return {?}
     */
    RangerSildersComponent.prototype.writeValue = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        if (value !== undefined) {
            this.value = value;
        }
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    RangerSildersComponent.prototype.registerOnChange = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.propagateChange = fn;
    };
    /**
     * @return {?}
     */
    RangerSildersComponent.prototype.registerOnTouched = /**
     * @return {?}
     */
    function () { };
    RangerSildersComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'ranger-silders',
                    template: "<div class=\"rf-rangers\">\n  <section id=\"rf-range-slider-{{ id }}\" class=\"range-slider\">\n    <input value=\"{{ value.min }}\" min=\"{{ min }}\" max=\"{{ max }}\" step=\"{{ step }}\" type=\"range\" />\n    <input value=\"{{ value.max }}\" min=\"{{ min }}\" max=\"{{ max }}\" step=\"{{ step }}\" type=\"range\" />\n    <span class=\"rangeValues\"></span>\n  </section>\n</div>\n",
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef((/**
                             * @return {?}
                             */
                            function () { return RangerSildersComponent; })),
                            multi: true
                        }
                    ],
                    styles: [".rangeValues{position:absolute;top:30px;left:0}section.range-slider{position:relative;float:left;width:100%;text-align:center}section.range-slider input[type=range]{pointer-events:none;position:absolute;-webkit-appearance:none;-webkit-tap-highlight-color:rgba(255,255,255,0);border:none;border-radius:14px;background:#f1efef;box-shadow:inset 0 1px 0 0 #cdc6c6,inset 0 -1px 0 0 #d9d4d4;-webkit-box-shadow:inset 0 1px 0 0 #cdc6c6,inset 0 -1px 0 0 #d9d4d4;overflow:hidden;left:0;top:0;outline:0;height:5px;margin:0;padding:0;width:100%}section.range-slider input[type=range]::-webkit-slider-thumb{cursor:pointer;pointer-events:all;position:relative;z-index:9999;outline:0;-webkit-appearance:none;width:10px;height:10px;border:none;border-radius:50%;background-color:#4cd964}section.range-slider input[type=range]::-moz-range-thumb{cursor:pointer;pointer-events:all;position:relative;z-index:9999;-moz-appearance:none;width:10px;height:10px;border:none;border-radius:50%;background-color:#4cd964}section.range-slider input[type=range]::-ms-thumb{cursor:pointer;pointer-events:all;position:relative;z-index:9999;-ms-appearance:none;width:10px;height:10px;border-radius:50%;border:0;background-color:#4cd964}section.range-slider input[type=range]::-moz-range-track{position:relative;z-index:-1;background-color:#000;border:0}section.range-slider input[type=range]:last-of-type::-moz-range-track{-moz-appearance:none;background:none;border:0}section.range-slider input[type=range]::-moz-focus-outer{border:0}"]
                }] }
    ];
    /** @nocollapse */
    RangerSildersComponent.ctorParameters = function () { return []; };
    RangerSildersComponent.propDecorators = {
        id: [{ type: Input }],
        min: [{ type: Input }],
        miLabel: [{ type: Input }],
        unit: [{ type: Input }],
        max: [{ type: Input }],
        step: [{ type: Input }],
        _value: [{ type: Input }],
        oldValue: [{ type: Input }]
    };
    return RangerSildersComponent;
}());
export { RangerSildersComponent };
if (false) {
    /** @type {?} */
    RangerSildersComponent.prototype.id;
    /** @type {?} */
    RangerSildersComponent.prototype.min;
    /** @type {?} */
    RangerSildersComponent.prototype.miLabel;
    /** @type {?} */
    RangerSildersComponent.prototype.unit;
    /** @type {?} */
    RangerSildersComponent.prototype.max;
    /** @type {?} */
    RangerSildersComponent.prototype.step;
    /** @type {?} */
    RangerSildersComponent.prototype._value;
    /** @type {?} */
    RangerSildersComponent.prototype.oldValue;
    /** @type {?} */
    RangerSildersComponent.prototype.propagateChange;
}
