// @vicoders/reactive-form/esm5/lib/directives/custom-checkbox/custom-checkbox.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/custom-checkbox/custom-checkbox.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, forwardRef, ElementRef } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
var CustomCheckboxComponent = /** @class */ (function () {
    function CustomCheckboxComponent(element) {
        this.element = element;
        this.onChang = (/**
         * @param {?} val
         * @return {?}
         */
        function (val) { });
        this.onTouched = (/**
         * @return {?}
         */
        function () { });
        this.disabled = false;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    CustomCheckboxComponent.prototype.writeValue = /**
     * @param {?} val
     * @return {?}
     */
    function (val) {
        if (val) {
            this.value = val;
        }
    };
    Object.defineProperty(CustomCheckboxComponent.prototype, "value", {
        get: /**
         * @return {?}
         */
        function () {
            return this._value;
        },
        set: /**
         * @param {?} val
         * @return {?}
         */
        function (val) {
            this._value = val;
            this.onChang(this._value);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} fn
     * @return {?}
     */
    CustomCheckboxComponent.prototype.registerOnChange = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.onChang = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    CustomCheckboxComponent.prototype.registerOnTouched = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.onTouched = fn;
    };
    /**
     * @param {?} isDisabled
     * @return {?}
     */
    CustomCheckboxComponent.prototype.setDisabledState = /**
     * @param {?} isDisabled
     * @return {?}
     */
    function (isDisabled) {
        this.disabled = isDisabled;
    };
    /**
     * @return {?}
     */
    CustomCheckboxComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.value = false;
    };
    /**
     * @return {?}
     */
    CustomCheckboxComponent.prototype.onChangeFunction = /**
     * @return {?}
     */
    function () {
        this.onChange();
    };
    CustomCheckboxComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'custom-checkbox',
                    template: "<input type=\"checkbox\" [(ngModel)]=\"value\" (change)=\"onChangeFunction()\">\n",
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef((/**
                             * @return {?}
                             */
                            function () { return CustomCheckboxComponent; })),
                            multi: true
                        }
                    ],
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    CustomCheckboxComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    CustomCheckboxComponent.propDecorators = {
        _value: [{ type: Input }],
        onChange: [{ type: Input }]
    };
    return CustomCheckboxComponent;
}());
export { CustomCheckboxComponent };
if (false) {
    /** @type {?} */
    CustomCheckboxComponent.prototype._value;
    /**
     * @type {?}
     * @private
     */
    CustomCheckboxComponent.prototype.disabled;
    /**
     * @type {?}
     * @private
     */
    CustomCheckboxComponent.prototype.onChang;
    /**
     * @type {?}
     * @private
     */
    CustomCheckboxComponent.prototype.onTouched;
    /** @type {?} */
    CustomCheckboxComponent.prototype.onChange;
    /**
     * @type {?}
     * @private
     */
    CustomCheckboxComponent.prototype.element;
}
