// @vicoders/reactive-form/esm2015/lib/directives/gutenburg/gutenburg.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/gutenburg/gutenburg.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, forwardRef, Input } from '@angular/core';
// import { IBlock, BlockEditorProviderComponent } from '@vicoders/vc-gutenberg';
// import { serialize, parse } from '@wordpress/blocks';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
export class GutenburgComponent {
    constructor() {
        this.propagateChange = (/**
         * @param {?} val
         * @return {?}
         */
        (val) => { });
        // this.blocks = [];
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        setTimeout((/**
         * @return {?}
         */
        () => {
            if (this.oldValue) {
                // this.blocks = parse(this.oldValue);
                this.value = this.oldValue;
            }
        }), 1000);
    }
    // onBlocksChanged(item: IBlock[]) {
    //   const html = serialize(item);
    //   this.value = html;
    //   console.log('value', this.value);
    // }
    /**
     * @return {?}
     */
    get value() {
        return this._value;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    set value(val) {
        this._value = val;
        this.propagateChange(this._value);
    }
    /**
     * @param {?} value
     * @return {?}
     */
    writeValue(value) {
        if (value !== undefined) {
            this.value = value;
        }
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnChange(fn) {
        this.propagateChange = fn;
    }
    /**
     * @return {?}
     */
    registerOnTouched() { }
    /**
     * @return {?}
     */
    ngOnInit() { }
}
GutenburgComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'gutenberg',
                template: "<!-- <div class=\"form-group\">\n  <block-editor-provider\n    #editor\n    *ngIf=\"blocks\"\n    [blocks]=\"blocks\"\n    (blocksChange)=\"onBlocksChanged($event)\"\n    contentClass=\"gutenberg__editor\"\n  ></block-editor-provider>\n</div> -->\n",
                providers: [
                    {
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef((/**
                         * @return {?}
                         */
                        () => GutenburgComponent)),
                        multi: true
                    }
                ],
                styles: [""]
            }] }
];
/** @nocollapse */
GutenburgComponent.ctorParameters = () => [];
GutenburgComponent.propDecorators = {
    _value: [{ type: Input }],
    oldValue: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    GutenburgComponent.prototype._value;
    /** @type {?} */
    GutenburgComponent.prototype.oldValue;
    /** @type {?} */
    GutenburgComponent.prototype.propagateChange;
}
