// @vicoders/reactive-form/esm2015/lib/angular-reactive-form.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/angular-reactive-form.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule, Injector } from '@angular/core';
import { AngularReactiveFormComponent } from './angular-reactive-form.component';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { DirectivesModule } from './directives/directives.module';
import { EditorModule } from '@tinymce/tinymce-angular';
import { ReactiveFormComponent } from './components/reactive-form/reactive-form.component';
import { DynamicInputComponent } from './components/dynamic-input/dynamic-input.component';
import { setAppInjector } from './app-injector';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { PipesModule } from './pipes/pipes.module';
/** @type {?} */
export const UPLOAD_DIRECTIVE_HEADERS = 'UPLOAD_DIRECTIVE_HEADERS';
/** @type {?} */
export const UPLOAD_DIRECTIVE_API_URL = 'UPLOAD_DIRECTIVE_API_URL';
/** @type {?} */
export const TINYMCE_CONFIG = 'TINYMCE_CONFIG';
const ɵ0 = {
    example: 'example'
}, ɵ1 = {
    apiUpload: 'example',
    paramName: 'example',
    resultTransformer: 'example'
};
export class AngularReactiveFormModule {
    /**
     * @param {?} injector
     */
    constructor(injector) {
        setAppInjector(injector);
    }
}
AngularReactiveFormModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    ReactiveFormsModule,
                    PipesModule,
                    DirectivesModule,
                    EditorModule,
                    NgbModule
                ],
                declarations: [AngularReactiveFormComponent, ReactiveFormComponent, DynamicInputComponent],
                exports: [AngularReactiveFormComponent, ReactiveFormComponent],
                providers: [
                    {
                        provide: UPLOAD_DIRECTIVE_HEADERS,
                        useValue: ɵ0
                    },
                    {
                        provide: UPLOAD_DIRECTIVE_API_URL,
                        useValue: 'example'
                    },
                    {
                        provide: TINYMCE_CONFIG,
                        useValue: ɵ1
                    }
                ]
            },] }
];
/** @nocollapse */
AngularReactiveFormModule.ctorParameters = () => [
    { type: Injector }
];
export { ɵ0, ɵ1 };
