// @vicoders/reactive-form/esm2015/lib/directives/mutiple-checkbox/mutiple-checkbox.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/mutiple-checkbox/mutiple-checkbox.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, forwardRef, ElementRef } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import * as _ from 'lodash';
export class MutipleCheckboxComponent {
    /**
     * @param {?} element
     */
    constructor(element) {
        this.element = element;
        this.checkedList = [];
        this.onChange = (/**
         * @param {?} val
         * @return {?}
         */
        (val) => { });
        this.onTouched = (/**
         * @return {?}
         */
        () => { });
        this.disabled = false;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    writeValue(val) {
        if (val) {
            this.value = val;
        }
    }
    /**
     * @return {?}
     */
    get value() {
        return this._value;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    set value(val) {
        this.selected = _.find(this.options, (/**
         * @param {?} item
         * @return {?}
         */
        item => item.value === val));
        this._value = val;
        this.onChange(this._value);
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnChange(fn) {
        this.onChange = fn;
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    /**
     * @param {?} isDisabled
     * @return {?}
     */
    setDisabledState(isDisabled) {
        this.disabled = isDisabled;
    }
    // tslint:disable-next-line:use-life-cycle-interface
    /**
     * @return {?}
     */
    ngOnInit() { }
    /**
     * @return {?}
     */
    ngOnChanges() {
        setTimeout((/**
         * @return {?}
         */
        () => {
            if (this.isClear === true) {
                if (!_.isUndefined(this.options) && !_.isEmpty(this.options)) {
                    _.map(this.options, (/**
                     * @param {?} i
                     * @return {?}
                     */
                    i => {
                        i.checked = undefined;
                    }));
                    this.checkedList = [];
                    this.value = this.checkedList;
                }
            }
        }), 2);
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        setTimeout((/**
         * @return {?}
         */
        () => {
            if (this.options !== null && this.options !== undefined && !_.isEmpty(this.options)) {
                /** @type {?} */
                const $this = this;
                _.forEach(this.options, (/**
                 * @param {?} value
                 * @return {?}
                 */
                function (value) {
                    if (value.checked === true) {
                        $this.checkedList.push(value.value);
                    }
                }));
                this.value = this.checkedList;
            }
            this.value = this.checkedList;
        }), 1);
    }
    /**
     * @param {?} option
     * @param {?} event
     * @return {?}
     */
    onCheckboxChange(option, event) {
        if (event.target.checked) {
            this.checkedList.push(option.value);
        }
        else {
            for (let i = 0; i < this.options.length; i++) {
                if (this.checkedList[i] === option.value) {
                    this.checkedList.splice(i, 1);
                }
            }
        }
        this.value = this.checkedList;
    }
}
MutipleCheckboxComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'mutiple-checkbox',
                template: "<div class=\"checkbox\" *ngFor=\"let list of options\">\n  <!-- <label *ngIf=\"isDisabled\">\n    <input\n      disabled\n      type=\"checkbox\"\n      value=\"{{ list.value }}\"\n      [attr.checked]=\"list.checked === true ? 'checked' : null\"\n      (change)=\"onCheckboxChange(list, $event)\"\n    />{{ list.label }}</label\n  >\n  <label *ngIf=\"!isDisabled\">\n    <input\n      type=\"checkbox\"\n      value=\"{{ list.value }}\"\n      [attr.checked]=\"list.checked === true ? 'checked' : null\"\n      (change)=\"onCheckboxChange(list, $event)\"\n    />{{ list.label }}</label\n  > -->\n  <label class=\"float-label\" *ngIf=\"isDisabled\"\n    >{{ list.label }}\n    <input\n      disabled\n      type=\"checkbox\"\n      value=\"{{ list.value }}\"\n      [attr.checked]=\"list.checked === true ? 'checked' : null\"\n      (change)=\"onCheckboxChange(list, $event)\"\n    />\n    <span class=\"checkmark\"></span>\n  </label>\n  <label class=\"float-label\" *ngIf=\"!isDisabled\"\n    >{{ list.label }}\n    <input\n      type=\"checkbox\"\n      value=\"{{ list.value }}\"\n      [attr.checked]=\"list.checked === true ? 'checked' : null\"\n      (change)=\"onCheckboxChange(list, $event)\"\n    />\n    <span class=\"checkmark\"></span>\n  </label>\n</div>\n",
                providers: [
                    {
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef((/**
                         * @return {?}
                         */
                        () => MutipleCheckboxComponent)),
                        multi: true
                    }
                ],
                styles: [".checkbox .float-label{display:block;position:relative;padding-left:35px;margin-bottom:12px;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;line-height:27px}.checkbox .float-label input{position:absolute;opacity:0;cursor:pointer;height:0;width:0}.checkbox .checkmark{position:absolute;top:0;left:0;height:25px;width:25px;background-color:#eee}.checkbox .float-label:hover input~.checkmark{background-color:#ccc}.checkbox .float-label input:checked~.checkmark{background-color:#2196f3}.checkbox .checkmark:after{content:'';position:absolute;display:none}.checkbox .float-label input:checked~.checkmark:after{display:block}.checkbox .float-label .checkmark:after{left:9px;top:5px;width:5px;height:10px;border:solid #fff;border-width:0 3px 3px 0;-webkit-transform:rotate(45deg);transform:rotate(45deg)}"]
            }] }
];
/** @nocollapse */
MutipleCheckboxComponent.ctorParameters = () => [
    { type: ElementRef }
];
MutipleCheckboxComponent.propDecorators = {
    options: [{ type: Input }],
    _value: [{ type: Input }],
    isDisabled: [{ type: Input }],
    isClear: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    MutipleCheckboxComponent.prototype.options;
    /** @type {?} */
    MutipleCheckboxComponent.prototype._value;
    /** @type {?} */
    MutipleCheckboxComponent.prototype.isDisabled;
    /** @type {?} */
    MutipleCheckboxComponent.prototype.isClear;
    /** @type {?} */
    MutipleCheckboxComponent.prototype.checkedList;
    /** @type {?} */
    MutipleCheckboxComponent.prototype.selected;
    /**
     * @type {?}
     * @private
     */
    MutipleCheckboxComponent.prototype.disabled;
    /**
     * @type {?}
     * @private
     */
    MutipleCheckboxComponent.prototype.onChange;
    /**
     * @type {?}
     * @private
     */
    MutipleCheckboxComponent.prototype.onTouched;
    /**
     * @type {?}
     * @private
     */
    MutipleCheckboxComponent.prototype.element;
}
