// @vicoders/reactive-form/esm2015/lib/directives/captcha/captcha.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/captcha/captcha.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, forwardRef, Input } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
export class CaptchaComponent {
    constructor() {
        this._captchaData = {
            value: '',
            code: ''
        };
        this.propagateChange = (/**
         * @param {?} val
         * @return {?}
         */
        (val) => { });
    }
    /**
     * @return {?}
     */
    ngOnInit() { }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        setTimeout((/**
         * @return {?}
         */
        () => {
            this.captchaData = Object.assign({}, this.captchaData, { code: this.random() });
        }), 1);
    }
    /**
     * @return {?}
     */
    get captchaData() {
        return this._captchaData;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    set captchaData(val) {
        this._captchaData = val;
        this.propagateChange(this._captchaData);
    }
    /**
     * @param {?} value
     * @return {?}
     */
    writeValue(value) {
        if (value !== undefined) {
            this.captchaData = value;
        }
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnChange(fn) {
        this.propagateChange = fn;
    }
    /**
     * @return {?}
     */
    registerOnTouched() { }
    /**
     * @return {?}
     */
    randIntWithZero() {
        return Math.floor(Math.random() * 10);
    }
    /**
     * @return {?}
     */
    random() {
        /** @type {?} */
        let code = '';
        /** @type {?} */
        let characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
        if (this.type === 'number') {
            characters = '0123456789';
            // for (let i = 0; i < this.numberOfChar; i++) {
            //   code += this.randIntWithZero();
            // }
        }
        else if (this.type === 'alphabet') {
            characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
        }
        /** @type {?} */
        const charactersLength = characters.length;
        for (let i = 0; i < this.numberOfChar; i++) {
            code += characters.charAt(Math.floor(Math.random() * charactersLength));
        }
        return code;
    }
    /**
     * @return {?}
     */
    randomCode() {
        this.captchaData = Object.assign({}, this.captchaData, { code: this.random() });
    }
}
CaptchaComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'captcha',
                template: "<div class=\"captch--rform\">\n  <input class=\"form-control captch--controll\" maxlength=\"{{ numberOfChar }}\" type=\"text\" [(ngModel)]=\"captchaData.value\" />\n  <span *ngIf=\"captchaData.code\">{{ captchaData.code }}</span>\n  <button type=\"button\" (click)=\"randomCode()\"><i class=\"fa fa-refresh\" aria-hidden=\"true\"></i></button>\n  <p class=\"rf__captcha--error\" *ngIf=\"!captchaCorrect\">{{ message }}</p>\n</div>\n",
                providers: [
                    {
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef((/**
                         * @return {?}
                         */
                        () => CaptchaComponent)),
                        multi: true
                    }
                ],
                styles: [".captch--rform span{font-family:lato;font-weight:700;letter-spacing:2px;font-style:italic;margin:0 15px}.captch--rform button{cursor:pointer}.captch--rform .captch--controll{display:inline-block;width:200px}.captch--rform .rf__captcha--error{margin-top:15px;color:#e74c3c;font-size:15px}"]
            }] }
];
/** @nocollapse */
CaptchaComponent.ctorParameters = () => [];
CaptchaComponent.propDecorators = {
    _captchaData: [{ type: Input }],
    numberOfChar: [{ type: Input }],
    type: [{ type: Input }],
    captchaCorrect: [{ type: Input }],
    message: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    CaptchaComponent.prototype._captchaData;
    /** @type {?} */
    CaptchaComponent.prototype.numberOfChar;
    /** @type {?} */
    CaptchaComponent.prototype.type;
    /** @type {?} */
    CaptchaComponent.prototype.captchaCorrect;
    /** @type {?} */
    CaptchaComponent.prototype.message;
    /** @type {?} */
    CaptchaComponent.prototype.propagateChange;
}
