// @vicoders/reactive-form/esm5/lib/directives/custom-selection/custom-selection.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/custom-selection/custom-selection.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, forwardRef, ElementRef, Renderer2 } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import * as _ from 'lodash';
var CustomSelectionComponent = /** @class */ (function () {
    function CustomSelectionComponent(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.onChange = (/**
         * @param {?} val
         * @return {?}
         */
        function (val) { });
        this.onTouched = (/**
         * @return {?}
         */
        function () { });
        this.disabled = false;
    }
    /**
     * @return {?}
     */
    CustomSelectionComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    /**
     * @return {?}
     */
    CustomSelectionComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        setTimeout((/**
         * @return {?}
         */
        function () {
            if (_this.oldValue) {
                _this.selected = _.find(_this.options, (/**
                 * @param {?} item
                 * @return {?}
                 */
                function (item) { return item.value === _this.oldValue.value; }));
                _this._value = _this.oldValue;
                _this.onChange(_this._value);
            }
        }), 1);
    };
    /**
     * @return {?}
     */
    CustomSelectionComponent.prototype.triggerChanged = /**
     * @return {?}
     */
    function () {
        // tslint:disable-next-line:prefer-const
        /** @type {?} */
        var event = new CustomEvent('change', { bubbles: true });
        ((/** @type {?} */ (this.renderer))).invokeElementMethod(this.element.nativeElement, 'dispatchEvent', [event]);
    };
    /**
     * @param {?} val
     * @return {?}
     */
    CustomSelectionComponent.prototype.writeValue = /**
     * @param {?} val
     * @return {?}
     */
    function (val) {
        if (val) {
            this.value = val;
        }
    };
    Object.defineProperty(CustomSelectionComponent.prototype, "value", {
        get: /**
         * @return {?}
         */
        function () {
            return this._value;
        },
        set: /**
         * @param {?} val
         * @return {?}
         */
        function (val) {
            this.selected = _.find(this.options, (/**
             * @param {?} item
             * @return {?}
             */
            function (item) { return item.value === val.value; }));
            this._value = val;
            this.onChange(this._value);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} fn
     * @return {?}
     */
    CustomSelectionComponent.prototype.registerOnChange = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.onChange = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    CustomSelectionComponent.prototype.registerOnTouched = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.onTouched = fn;
    };
    /**
     * @param {?} isDisabled
     * @return {?}
     */
    CustomSelectionComponent.prototype.setDisabledState = /**
     * @param {?} isDisabled
     * @return {?}
     */
    function (isDisabled) {
        this.disabled = isDisabled;
    };
    /**
     * @param {?} opt
     * @return {?}
     */
    CustomSelectionComponent.prototype.change = /**
     * @param {?} opt
     * @return {?}
     */
    function (opt) {
        this.value = opt;
        this.triggerChanged();
    };
    CustomSelectionComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'custom-selection',
                    template: "<div class=\"dropdown custom-selection\">\n  <button class=\"btn btn-default dropdown-toggle\" type=\"button\" data-toggle=\"dropdown\" aria-haspopup=\"true\" aria-expanded=\"false\">\n    {{ selected === undefined ? '' : selected.label }}\n  </button>\n  <div class=\"dropdown-menu\">\n    <a class=\"dropdown-item\" *ngFor=\"let option of options\" (click)=\"change(option)\">{{ option.label }}</a>\n  </div>\n</div>\n",
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef((/**
                             * @return {?}
                             */
                            function () { return CustomSelectionComponent; })),
                            multi: true
                        }
                    ],
                    styles: [".custom-selection .dropdown-menu .dropdown-item{cursor:pointer}"]
                }] }
    ];
    /** @nocollapse */
    CustomSelectionComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 }
    ]; };
    CustomSelectionComponent.propDecorators = {
        _value: [{ type: Input }],
        options: [{ type: Input }],
        oldValue: [{ type: Input }]
    };
    return CustomSelectionComponent;
}());
export { CustomSelectionComponent };
if (false) {
    /** @type {?} */
    CustomSelectionComponent.prototype._value;
    /** @type {?} */
    CustomSelectionComponent.prototype.options;
    /** @type {?} */
    CustomSelectionComponent.prototype.oldValue;
    /** @type {?} */
    CustomSelectionComponent.prototype.selected;
    /**
     * @type {?}
     * @private
     */
    CustomSelectionComponent.prototype.disabled;
    /**
     * @type {?}
     * @private
     */
    CustomSelectionComponent.prototype.onChange;
    /**
     * @type {?}
     * @private
     */
    CustomSelectionComponent.prototype.onTouched;
    /**
     * @type {?}
     * @private
     */
    CustomSelectionComponent.prototype.element;
    /**
     * @type {?}
     * @private
     */
    CustomSelectionComponent.prototype.renderer;
}
