// @vicoders/reactive-form/esm5/lib/directives/gutenburg/gutenburg.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/gutenburg/gutenburg.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, forwardRef, Input } from '@angular/core';
// import { IBlock, BlockEditorProviderComponent } from '@vicoders/vc-gutenberg';
// import { serialize, parse } from '@wordpress/blocks';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
var GutenburgComponent = /** @class */ (function () {
    function GutenburgComponent() {
        this.propagateChange = (/**
         * @param {?} val
         * @return {?}
         */
        function (val) { });
        // this.blocks = [];
    }
    /**
     * @return {?}
     */
    GutenburgComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        setTimeout((/**
         * @return {?}
         */
        function () {
            if (_this.oldValue) {
                // this.blocks = parse(this.oldValue);
                _this.value = _this.oldValue;
            }
        }), 1000);
    };
    Object.defineProperty(GutenburgComponent.prototype, "value", {
        // onBlocksChanged(item: IBlock[]) {
        //   const html = serialize(item);
        //   this.value = html;
        //   console.log('value', this.value);
        // }
        get: 
        // onBlocksChanged(item: IBlock[]) {
        //   const html = serialize(item);
        //   this.value = html;
        //   console.log('value', this.value);
        // }
        /**
         * @return {?}
         */
        function () {
            return this._value;
        },
        set: /**
         * @param {?} val
         * @return {?}
         */
        function (val) {
            this._value = val;
            this.propagateChange(this._value);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} value
     * @return {?}
     */
    GutenburgComponent.prototype.writeValue = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        if (value !== undefined) {
            this.value = value;
        }
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    GutenburgComponent.prototype.registerOnChange = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.propagateChange = fn;
    };
    /**
     * @return {?}
     */
    GutenburgComponent.prototype.registerOnTouched = /**
     * @return {?}
     */
    function () { };
    /**
     * @return {?}
     */
    GutenburgComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    GutenburgComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'gutenberg',
                    template: "<!-- <div class=\"form-group\">\n  <block-editor-provider\n    #editor\n    *ngIf=\"blocks\"\n    [blocks]=\"blocks\"\n    (blocksChange)=\"onBlocksChanged($event)\"\n    contentClass=\"gutenberg__editor\"\n  ></block-editor-provider>\n</div> -->\n",
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef((/**
                             * @return {?}
                             */
                            function () { return GutenburgComponent; })),
                            multi: true
                        }
                    ],
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    GutenburgComponent.ctorParameters = function () { return []; };
    GutenburgComponent.propDecorators = {
        _value: [{ type: Input }],
        oldValue: [{ type: Input }]
    };
    return GutenburgComponent;
}());
export { GutenburgComponent };
if (false) {
    /** @type {?} */
    GutenburgComponent.prototype._value;
    /** @type {?} */
    GutenburgComponent.prototype.oldValue;
    /** @type {?} */
    GutenburgComponent.prototype.propagateChange;
}
