// @vicoders/reactive-form/esm5/lib/directives/province-city/province-city.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/province-city/province-city.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, forwardRef } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { Provinces } from './province';
var ProvinceCityComponent = /** @class */ (function () {
    function ProvinceCityComponent() {
        this.provinces = Provinces;
        this.propagateChange = (/**
         * @param {?} val
         * @return {?}
         */
        function (val) { });
    }
    /**
     * @return {?}
     */
    ProvinceCityComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    Object.defineProperty(ProvinceCityComponent.prototype, "value", {
        get: /**
         * @return {?}
         */
        function () {
            return this._value;
        },
        set: /**
         * @param {?} val
         * @return {?}
         */
        function (val) {
            this._value = val;
            this.propagateChange(this._value);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} value
     * @return {?}
     */
    ProvinceCityComponent.prototype.selectedProvince = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.value = value;
    };
    /**
     * @param {?} value
     * @return {?}
     */
    ProvinceCityComponent.prototype.writeValue = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        if (value !== undefined) {
            this.value = value;
        }
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ProvinceCityComponent.prototype.registerOnChange = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.propagateChange = fn;
    };
    /**
     * @return {?}
     */
    ProvinceCityComponent.prototype.registerOnTouched = /**
     * @return {?}
     */
    function () { };
    ProvinceCityComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'province-city',
                    template: "<div class=\"form-group\">\n  <select class=\"form-control\" #province (change)=\"selectedProvince(province.value)\" [(ngModel)]=\"value\">\n    <option *ngFor=\"let item of provinces\" value=\"{{item.name}}\">{{item.name}}</option>\n  </select>\n</div>\n",
                    providers: [{
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef((/**
                             * @return {?}
                             */
                            function () { return ProvinceCityComponent; })),
                            multi: true
                        }],
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    ProvinceCityComponent.ctorParameters = function () { return []; };
    ProvinceCityComponent.propDecorators = {
        _value: [{ type: Input }]
    };
    return ProvinceCityComponent;
}());
export { ProvinceCityComponent };
if (false) {
    /** @type {?} */
    ProvinceCityComponent.prototype.provinces;
    /** @type {?} */
    ProvinceCityComponent.prototype._value;
    /** @type {?} */
    ProvinceCityComponent.prototype.propagateChange;
}
