// @vicoders/reactive-form/esm5/lib/directives/typeahead-without-api/typeahead.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/typeahead-without-api/typeahead.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, forwardRef, Input, ViewChild } from '@angular/core';
import { SelectionService } from '../selection-by-api/selection-by-api';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { NgbTypeahead } from '@ng-bootstrap/ng-bootstrap';
import { Subject, merge } from 'rxjs';
import { debounceTime, distinctUntilChanged, filter, map } from 'rxjs/operators';
var TypeaheadWithoutApiComponent = /** @class */ (function () {
    function TypeaheadWithoutApiComponent() {
        var _this = this;
        this.states = [];
        this.focus$ = new Subject();
        this.click$ = new Subject();
        this.formatter = (/**
         * @param {?} x
         * @return {?}
         */
        function (x) { return x.label; });
        this.search = (/**
         * @param {?} text$
         * @return {?}
         */
        function (text$) {
            /** @type {?} */
            var debouncedText$ = text$.pipe(debounceTime(200), distinctUntilChanged());
            /** @type {?} */
            var clicksWithClosedPopup$ = _this.click$.pipe(filter((/**
             * @return {?}
             */
            function () { return !_this.instance.isPopupOpen(); })));
            /** @type {?} */
            var inputFocus$ = _this.focus$;
            return merge(debouncedText$, inputFocus$, clicksWithClosedPopup$).pipe(
            // tslint:disable-next-line:max-line-length
            map((/**
             * @param {?} term
             * @return {?}
             */
            function (term) {
                return (term === '' ? _this.states : _this.states.filter((/**
                 * @param {?} v
                 * @return {?}
                 */
                function (v) { return v.label.toLowerCase().indexOf(term.toLowerCase()) > -1; }))).slice(0, 10);
            })));
            // tslint:disable-next-line:semicolon
        });
        this.propagateChange = (/**
         * @param {?} val
         * @return {?}
         */
        function (val) { });
    }
    /**
     * @return {?}
     */
    TypeaheadWithoutApiComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    /**
     * @return {?}
     */
    TypeaheadWithoutApiComponent.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        var _this = this;
        setTimeout((/**
         * @return {?}
         */
        function () {
            if (_this.isClear) {
                _this.value = null;
                _this.model = _this.value;
            }
        }), 1);
    };
    /**
     * @param {?} data
     * @return {?}
     */
    TypeaheadWithoutApiComponent.prototype.filterData = /**
     * @param {?} data
     * @return {?}
     */
    function (data) {
        if (typeof data === 'object') {
            this.value = data;
        }
    };
    /**
     * @return {?}
     */
    TypeaheadWithoutApiComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        setTimeout((/**
         * @return {?}
         */
        function () {
            _this.states = _this.options;
            if (_this.value) {
                _this.model = _this.value;
            }
        }), 1);
    };
    Object.defineProperty(TypeaheadWithoutApiComponent.prototype, "value", {
        get: /**
         * @return {?}
         */
        function () {
            return this._value;
        },
        set: /**
         * @param {?} val
         * @return {?}
         */
        function (val) {
            this._value = val;
            this.propagateChange(this._value);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} value
     * @return {?}
     */
    TypeaheadWithoutApiComponent.prototype.writeValue = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        if (value !== undefined) {
            this.value = value;
        }
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    TypeaheadWithoutApiComponent.prototype.registerOnChange = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.propagateChange = fn;
    };
    /**
     * @return {?}
     */
    TypeaheadWithoutApiComponent.prototype.registerOnTouched = /**
     * @return {?}
     */
    function () { };
    TypeaheadWithoutApiComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'typeahead-without-api',
                    template: "<div>\n  <ng-template #rt let-r=\"result\" let-t=\"term\">\n    <ngb-highlight [result]=\"r.label\" [term]=\"t\"></ngb-highlight>\n  </ng-template>\n  <input\n    [resultTemplate]=\"rt\"\n    [inputFormatter]=\"formatter\"\n    id=\"typeahead-focus-{{ id }}\"\n    type=\"text\"\n    class=\"form-control\"\n    [(ngModel)]=\"model\"\n    [ngbTypeahead]=\"search\"\n    (focus)=\"focus$.next($event.target.value)\"\n    (click)=\"click$.next($event.target.value)\"\n    #instance=\"ngbTypeahead\"\n    (ngModelChange)=\"filterData(model)\"\n  />\n</div>\n",
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef((/**
                             * @return {?}
                             */
                            function () { return TypeaheadWithoutApiComponent; })),
                            multi: true
                        },
                        SelectionService
                    ],
                    styles: ["ngb-highlight{position:relative;margin-right:15px}ngb-highlight:after{position:absolute;content:'-';top:0;right:-10px}ngb-highlight:last-child:after{position:absolute;content:''}"]
                }] }
    ];
    /** @nocollapse */
    TypeaheadWithoutApiComponent.ctorParameters = function () { return []; };
    TypeaheadWithoutApiComponent.propDecorators = {
        _value: [{ type: Input }],
        id: [{ type: Input }],
        options: [{ type: Input }],
        isClear: [{ type: Input }],
        instance: [{ type: ViewChild, args: ['instance', {},] }]
    };
    return TypeaheadWithoutApiComponent;
}());
export { TypeaheadWithoutApiComponent };
if (false) {
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype._value;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.id;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.options;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.isClear;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.states;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.model;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.instance;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.focus$;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.click$;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.formatter;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.search;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.propagateChange;
}
