// @vicoders/reactive-form/esm2015/lib/directives/province-city/province-city.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/province-city/province-city.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, forwardRef } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { Provinces } from './province';
export class ProvinceCityComponent {
    constructor() {
        this.provinces = Provinces;
        this.propagateChange = (/**
         * @param {?} val
         * @return {?}
         */
        (val) => { });
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
    /**
     * @return {?}
     */
    get value() {
        return this._value;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    set value(val) {
        this._value = val;
        this.propagateChange(this._value);
    }
    /**
     * @param {?} value
     * @return {?}
     */
    selectedProvince(value) {
        this.value = value;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    writeValue(value) {
        if (value !== undefined) {
            this.value = value;
        }
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnChange(fn) {
        this.propagateChange = fn;
    }
    /**
     * @return {?}
     */
    registerOnTouched() { }
}
ProvinceCityComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'province-city',
                template: "<div class=\"form-group\">\n  <select class=\"form-control\" #province (change)=\"selectedProvince(province.value)\" [(ngModel)]=\"value\">\n    <option *ngFor=\"let item of provinces\" value=\"{{item.name}}\">{{item.name}}</option>\n  </select>\n</div>\n",
                providers: [{
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef((/**
                         * @return {?}
                         */
                        () => ProvinceCityComponent)),
                        multi: true
                    }],
                styles: [""]
            }] }
];
/** @nocollapse */
ProvinceCityComponent.ctorParameters = () => [];
ProvinceCityComponent.propDecorators = {
    _value: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    ProvinceCityComponent.prototype.provinces;
    /** @type {?} */
    ProvinceCityComponent.prototype._value;
    /** @type {?} */
    ProvinceCityComponent.prototype.propagateChange;
}
