// @vicoders/reactive-form/esm2015/lib/directives/typeahead-without-api/typeahead.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directives/typeahead-without-api/typeahead.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, forwardRef, Input, ViewChild } from '@angular/core';
import { SelectionService } from '../selection-by-api/selection-by-api';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { NgbTypeahead } from '@ng-bootstrap/ng-bootstrap';
import { Subject, merge } from 'rxjs';
import { debounceTime, distinctUntilChanged, filter, map } from 'rxjs/operators';
export class TypeaheadWithoutApiComponent {
    constructor() {
        this.states = [];
        this.focus$ = new Subject();
        this.click$ = new Subject();
        this.formatter = (/**
         * @param {?} x
         * @return {?}
         */
        x => x.label);
        this.search = (/**
         * @param {?} text$
         * @return {?}
         */
        (text$) => {
            /** @type {?} */
            const debouncedText$ = text$.pipe(debounceTime(200), distinctUntilChanged());
            /** @type {?} */
            const clicksWithClosedPopup$ = this.click$.pipe(filter((/**
             * @return {?}
             */
            () => !this.instance.isPopupOpen())));
            /** @type {?} */
            const inputFocus$ = this.focus$;
            return merge(debouncedText$, inputFocus$, clicksWithClosedPopup$).pipe(
            // tslint:disable-next-line:max-line-length
            map((/**
             * @param {?} term
             * @return {?}
             */
            term => (term === '' ? this.states : this.states.filter((/**
             * @param {?} v
             * @return {?}
             */
            v => v.label.toLowerCase().indexOf(term.toLowerCase()) > -1))).slice(0, 10))));
            // tslint:disable-next-line:semicolon
        });
        this.propagateChange = (/**
         * @param {?} val
         * @return {?}
         */
        (val) => { });
    }
    /**
     * @return {?}
     */
    ngOnInit() { }
    /**
     * @return {?}
     */
    ngOnChanges() {
        setTimeout((/**
         * @return {?}
         */
        () => {
            if (this.isClear) {
                this.value = null;
                this.model = this.value;
            }
        }), 1);
    }
    /**
     * @param {?} data
     * @return {?}
     */
    filterData(data) {
        if (typeof data === 'object') {
            this.value = data;
        }
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        setTimeout((/**
         * @return {?}
         */
        () => {
            this.states = this.options;
            if (this.value) {
                this.model = this.value;
            }
        }), 1);
    }
    /**
     * @return {?}
     */
    get value() {
        return this._value;
    }
    /**
     * @param {?} val
     * @return {?}
     */
    set value(val) {
        this._value = val;
        this.propagateChange(this._value);
    }
    /**
     * @param {?} value
     * @return {?}
     */
    writeValue(value) {
        if (value !== undefined) {
            this.value = value;
        }
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnChange(fn) {
        this.propagateChange = fn;
    }
    /**
     * @return {?}
     */
    registerOnTouched() { }
}
TypeaheadWithoutApiComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'typeahead-without-api',
                template: "<div>\n  <ng-template #rt let-r=\"result\" let-t=\"term\">\n    <ngb-highlight [result]=\"r.label\" [term]=\"t\"></ngb-highlight>\n  </ng-template>\n  <input\n    [resultTemplate]=\"rt\"\n    [inputFormatter]=\"formatter\"\n    id=\"typeahead-focus-{{ id }}\"\n    type=\"text\"\n    class=\"form-control\"\n    [(ngModel)]=\"model\"\n    [ngbTypeahead]=\"search\"\n    (focus)=\"focus$.next($event.target.value)\"\n    (click)=\"click$.next($event.target.value)\"\n    #instance=\"ngbTypeahead\"\n    (ngModelChange)=\"filterData(model)\"\n  />\n</div>\n",
                providers: [
                    {
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef((/**
                         * @return {?}
                         */
                        () => TypeaheadWithoutApiComponent)),
                        multi: true
                    },
                    SelectionService
                ],
                styles: ["ngb-highlight{position:relative;margin-right:15px}ngb-highlight:after{position:absolute;content:'-';top:0;right:-10px}ngb-highlight:last-child:after{position:absolute;content:''}"]
            }] }
];
/** @nocollapse */
TypeaheadWithoutApiComponent.ctorParameters = () => [];
TypeaheadWithoutApiComponent.propDecorators = {
    _value: [{ type: Input }],
    id: [{ type: Input }],
    options: [{ type: Input }],
    isClear: [{ type: Input }],
    instance: [{ type: ViewChild, args: ['instance', {},] }]
};
if (false) {
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype._value;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.id;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.options;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.isClear;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.states;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.model;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.instance;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.focus$;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.click$;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.formatter;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.search;
    /** @type {?} */
    TypeaheadWithoutApiComponent.prototype.propagateChange;
}
